src/burnchains/bitcoin/indexer.rs

```diff
@@ -385,47 +387,140 @@ impl BitcoinIndexer {
             .and_then(|_r| Ok(spv_client.end_block_height.unwrap()))
     }
 
+    #[cfg(test)]
+    fn new_reorg_spv_client(
+        reorg_headers_path: &str,
+        start_block: u64,
+        end_block: Option<u64>,
+        network_id: BitcoinNetworkType,
+    ) -> Result<SpvClient, btc_error> {
+        SpvClient::new_without_migration(
+            &reorg_headers_path,
+            start_block,
+            end_block,
+            network_id,
+            true,
+            true,
+        )
+    }
+
+    #[cfg(not(test))]
+    fn new_reorg_spv_client(
+        reorg_headers_path: &str,
+        start_block: u64,
+        end_block: Option<u64>,
+        network_id: BitcoinNetworkType,
+    ) -> Result<SpvClient, btc_error> {
+        SpvClient::new(
+            &reorg_headers_path,
+            start_block,
+            end_block,
+            network_id,
+            true,
+            true,
+        )
+    }
+
     /// Create a SPV client for starting reorg processing
     fn setup_reorg_headers(
         &mut self,
         canonical_spv_client: &SpvClient,
         reorg_headers_path: &str,
         start_block: u64,
+        remove_old: bool,
     ) -> Result<SpvClient, btc_error> {
-        if PathBuf::from(&reorg_headers_path).exists() {
-            fs::remove_file(&reorg_headers_path).map_err(|e| {
-                error!("Failed to remove {}", reorg_headers_path);
-                btc_error::Io(e)
-            })?;
+        if remove_old {
+            if PathBuf::from(&reorg_headers_path).exists() {
+                fs::remove_file(&reorg_headers_path).map_err(|e| {
+                    error!("Failed to remove {}", reorg_headers_path);
+                    btc_error::Io(e)
+                })?;
+            }
         }
 
         // bootstrap reorg client
-        let mut reorg_spv_client = SpvClient::new(
-            &reorg_headers_path,
+        let mut reorg_spv_client = BitcoinIndexer::new_reorg_spv_client(
+            reorg_headers_path,
             start_block,
             Some(start_block + REORG_BATCH_SIZE),
             self.runtime.network_id,
-            true,
-            true,
         )?;
+
         if start_block > 0 {
-            let start_header = canonical_spv_client
-                .read_block_header(start_block)?
-                .expect(&format!("BUG: missing block header for {}", start_block));
-            reorg_spv_client.insert_block_headers_before(start_block - 1, vec![start_header])?;
+            if start_block > BLOCK_DIFFICULTY_CHUNK_SIZE {
+                if remove_old {
+                    // set up a .reorg db
+                    // * needs the last difficulty interval of headers (note that the current
+                    // interval is `start_block / BLOCK_DIFFICULTY_CHUNK_SIZE - 1).
+                    // * needs the last interval's chain work calculation
+                    let interval_start_block = start_block / BLOCK_DIFFICULTY_CHUNK_SIZE - 2;
```

Comment:
What is `interval_start_block`? If the interval is 500, and `start_block` is 502, this is -2? Should that panic or underflow?

Reply:
This should be a saturating subtraction.